docs/_docs/04_reference/config-blocks-and-attributes.md

```diff
@@ -184,6 +184,9 @@ The `remote_state` block supports the following arguments:
   have support in Terragrunt to be automatically created if the storage does not exist. Currently `s3` and `gcs` are the
   two backends with support for automatic creation. Defaults to `false`.
 
+- `enable_dependency_optimization` (attribute): When `true`, attempt to optimize dependency fetching for terragrunt
```

Comment:
A bit hard to see this in the PR... Is this flag on the `remote_state` block or a global flag? If it's the former, do you need to put it on the `remote_state` block of the current module or *every module that is a dependency* of the current module?

Reply:
Talked with Yori live. Since most users likely have `remote_state` defined once, centrally, and `include`-ed everywhere else, this is effectively global.